File: clienMYExpert/src/app/classes/expert.ts
import { City } from './city';
import { Subject } from './subject';
import { User } from './user';
export class Expert {
  id: number;
  userName: string;
  userPassword: string;
  email: string;
  cityId: number;
  proSubject: number;
  businessName: string;
  description: string;
  imgUrl: string;
  scores: number;
  enable: boolean;
  phone: string;
  firstName: string;
  lastName: string;
  city?:City;
  subject?:Subject;
  numRecommends?:number;
  constructor(
    id?: number,
    userName?: string,
    userPassword?: string,
    email?: string,
    cityId?: number,
    proSubject?: number,
    img?: string,
    businessName?: string,
    description?: string,
    score?: number
  ) {
    this.id = id;
    this.userName = userName;
    this.userPassword = userPassword;
    this.email = email;
    this.cityId = cityId;
    this.proSubject = proSubject;
    this.imgUrl = img;
    this.businessName = businessName;
    this.description = description;
    this.scores = score;
    this.enable = true;
  }
}

File: Project/Project.DAL/Models/Expert.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

#nullable disable

namespace Project.DAL.Models
{
    public partial class Expert
    {
        public Expert()
        {
            Meetings = new HashSet<Meeting>();
            Recommends = new HashSet<Recommend>();
        }
        [Key]
        [Required]
        public int id { get; set; }
        [Required]
        public string userName { get; set; }
        [Required]
        public string userPassword { get; set; }
        [Required]
        public string email { get; set; }
        public string businessName { get; set; }
        public int cityId { get; set; }
        public string imgUrl { get; set; }
        public string phone { get; set; }
        [Required]
        public string firstName { get; set; }
        [Required]
        public string lastName { get; set; }
        public int proSubject { get; set; }
        public string description { get; set; }
        [Required]
        public bool enable { get; set; }
        [ForeignKey("cityId")]

        public virtual City City { get; set; }
        public virtual Subject Subject { get; set; }
        public virtual ICollection<Meeting> Meetings { get; set; }
        public virtual ICollection<Recommend> Recommends { get; set; }
    }
}
